Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -398,6 +399,36 @@ - (void)writeAvatarToDisk:(UIImage *)avatar
     });
 }
 
+- (NSData *)processedImageDataForRawAvatar:(UIImage *)image
+{
+    NSUInteger kMaxAvatarBytes = 5 * 1000 * 1000;
+
+    if (image.size.width != kOWSProfileManager_MaxAvatarWidth
```

Comment:
Since any scaling already induces a recompression, I decided this didn't warrant additional complexity. We might revisit this when doing manual cropping/scaling.